test/tools/unified-spec-runner/entities.ts

```diff
@@ -197,34 +198,44 @@ export class UnifiedMongoClient extends MongoClient {
     topology: 'MONGODB_LOG_TOPOLOGY'
   } as const;
 
-  constructor(uri: string, description: ClientEntity) {
-    const logCollector: { buffer: LogMessage[]; write: (log: Log) => void } = {
-      buffer: [],
-      write(log: Log): void {
-        const transformedLog = {
-          level: log.s,
-          component: log.c,
-          data: { ...log }
-        };
-
-        this.buffer.push(transformedLog);
-      }
-    };
-    const mongodbLogComponentSeverities = description.observeLogMessages;
-
-    super(uri, {
+  constructor(
+    uri: string,
+    description: ClientEntity,
+    config: {
+      loggingEnabled?: boolean;
+      setupLogging?: (options: Record<string, any>) => Record<string, any>;
+    }
+  ) {
+    const options: MongoClientOptions = {
       monitorCommands: true,
       __skipPingOnConnect: true,
-      mongodbLogComponentSeverities,
       ...getEnvironmentalOptions(),
       ...(description.serverApi ? { serverApi: description.serverApi } : {}),
-      mongodbLogPath: logCollector,
       // TODO(NODE-5785): We need to increase the truncation length because signature.hash is a Buffer making hellos too long
       mongodbLogMaxDocumentLength: 1250
-    } as any);
+    };
+
+    if (description.observeLogMessages != null) {
+      options.mongodbLogComponentSeverities = description.observeLogMessages;
+      options.mongodbLogPath = {
+        buffer: [],
+        write(log: Log): void {
+          const transformedLog = {
+            level: log.s,
+            component: log.c,
+            data: { ...log }
+          };
+
+          this.buffer.push(transformedLog);
+        }
+      } as MongoDBLogWritable;
+    } else if (config.loggingEnabled) {
+      config.setupLogging?.(options);
+    }
```

Comment:
Does this imply that any log tests cannot use the debug logging from the test configuration?

Reply:
Yes, we would have to filter the logs according the test settings to not collect more than requested. We can enhance this if there's a test that uses logging that flakes in the future.